src/bundles/pinning.js

```diff
@@ -140,33 +184,36 @@ const pinningBundle = {
     try {
       pinLocally ? await ipfs.pin.add(cid) : await ipfs.pin.rm(cid)
     } catch (e) {
-      console.error(e)
+      // its ok if unpinning failed because object is not pinned
+      if (e.message !== 'not pinned or pinned indirectly') {
+        console.error(`unexpected local pin error for ${cid} (${name})`, e)
+      }
     }
 
-    store.selectPinningServices().forEach(async service => {
+    const adds = []
+    const removals = []
+
+    store.selectPinningServices().filter(s => s.online).forEach(async service => {
       const shouldPin = services.includes(service.name)
       try {
+        const id = `${service.name}:${pin.cid}`
         if (shouldPin) {
-          dispatch({
-            type: 'ADD_REMOTE_PIN',
-            payload: {
-              ...pin,
-              id: `${service.name}:${pin.cid}`
-            }
-          })
-          await ipfs.pin.remote.add(cid, { service: service.name, name })
+          adds.push({ id, ...pin })
+          // TODO: remove background:true and add pin job to queue.
```

Comment:
Maybe let's add a github issue with this?

Reply:
Yes, filled https://github.com/ipfs/ipfs-webui/issues/1751 and https://github.com/ipfs/ipfs-webui/issues/1752
(we should add error handling first, tracking progress will be more complex)